src/ConstructionBase.jl

```diff
@@ -26,10 +26,7 @@ for (name, path) in [
     end
 end
 
-@generated function constructorof(::Type{T}) where T
-    getfield(parentmodule(T), nameof(T))
-end
-
+constructorof(T::Type) = Base.typename(T).wrapper
```

Comment:
```suggestion
constructorof(::Type{T}) where T = Base.typename(T).wrapper
```
Is there a reason why the `where T` was dropped as well?

Reply:
In practice, the where just reduces the information inference is permitted to use about the argument during inlining

Reply:
Ok, I guess I don't understand the scope of consequences of that. 

From practical experience using `where` on types and functions like this often fixes type stability bugs.